src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/AccessProfileServiceImpl.java

```diff
@@ -1,31 +1,35 @@
 package uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
-import java.util.stream.Collectors;
 import org.apache.commons.collections.CollectionUtils;
 import org.springframework.stereotype.Component;
 import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.AccessProfile;
 import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
 import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignmentFilteringResult;
-import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
 import uk.gov.hmcts.ccd.domain.model.definition.RoleToAccessProfileDefinition;
 import uk.gov.hmcts.ccd.domain.service.AccessControl;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
 @Component
 public class AccessProfileServiceImpl implements AccessProfileService, AccessControl {
 
     @Override
     public List<AccessProfile> generateAccessProfiles(RoleAssignmentFilteringResult filteringResults,
-                                                      CaseTypeDefinition caseTypeDefinition) {
+                                                      List<RoleToAccessProfileDefinition> roleToAccessProfiles) {
 
         List<AccessProfile> accessProfiles = new ArrayList<>();
+        Map<String, RoleToAccessProfileDefinition> roleToAccessProfileDefinitionMap =
+            toRoleNameAsKeyMap(roleToAccessProfiles);
```

Comment:
the map could have stayed in the CaseTypeDefinition, created once and for all by the setter